dipy/reconst/dti.py

```diff
@@ -841,31 +853,45 @@ def fit(self, data, mask=None, adjacency=False):
             *self.args,
             **self.kwargs,
         )
+
         if self.return_S0_hat:
             params_in_mask, model_S0 = params_in_mask
 
         if mask is None:
             out_shape = data.shape[:-1] + (-1,)
-            dti_params = params_in_mask.reshape(out_shape)
-            if self.return_S0_hat:
-                S0_params = model_S0.reshape(out_shape[:-1])
-            if extra is not None:
-                for key in extra:
-                    self.extra[key] = extra[key].reshape(data.shape)
+            if isinstance(params_in_mask, tuple):
```

Comment:
Made changes so, that it handle cases where `params_in_mask` is returned as a tuple instead of an array.

Reply:
Removed. This solution was incorrect.